kimchi/src/circuits/constraints.rs

```diff
@@ -668,23 +669,28 @@ impl<F: PrimeField + SquareRootField> Builder<F> {
                     },
                 )
                 .sum();
-            for runtime_table in runtime_tables.iter() {
-                num_lookups += runtime_table.len();
+            // After that on the runtime tables
+            if runtime_tables.is_some() {
+                let runtime_tables = runtime_tables.as_ref().unwrap();
+                for runtime_table in runtime_tables.iter() {
+                    lookup_domain_size += runtime_table.len();
+                }
             }
+            // And we add the built-in tables, depending on the features.
             let LookupFeatures { patterns, .. } = &lookup_features;
             for pattern in patterns.into_iter() {
```

Comment:
You may be interested in `Iterator.filter_map` for this cases.

Reply:
I do agree, maybe using instead a fold, but I would not add this in the scope of this PR.